src/views/StudentDetails.vue

```diff
@@ -0,0 +1,95 @@
+<script setup>
+import { ref, onMounted } from 'vue';
+import { useRoute } from 'vue-router';
+import { useStudentsStore } from '@/stores/students';
+import { useExercisesStore } from '@/stores/exercises';
+import { useGradesStore } from '@/stores/grades';
+import { storeToRefs } from 'pinia';
+
+const route = useRoute();
+const studentsStore = useStudentsStore();
+const exercisesStore = useExercisesStore();
+const gradesStore = useGradesStore();
+
+const studentId = route.params.id;
+const student = studentsStore.list.find((s) => s.id === studentId);
+const { list: exercises } = storeToRefs(exercisesStore);
+const avatarUrl = ref('');
+
+const fetchGithubProfilePicture = async () => {
+  if (student.githubUsername) {
+    try {
+      const response = await fetch(`https://api.github.com/users/${student.githubUsername}`);
+      if (response.ok) {
+        avatarUrl.value = (await response.json()).avatar_url;
+      } else {
+        console.error('User not found');
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+};
```

Comment:
What do you think about extracting all this logic to handle github avatar to it's own component?

Reply:
IMO it is okay to keep it like it is now because the feature I am implementing now (API integration) should bring student's data including avatar url, so this logic will end up being removed

However, in case this component should be added another sort of logic I totally agree on extracting it onto a new component